ProjectWizard uses deprecated functions - compilation warning

If possible, we should avoid using deprecated functions.
We now have this warning during compilation:

```
../app/projectwizard.cpp: In member function ‘QgsVectorLayer* ProjectWizard::createGpkgLayer(const QString&, const QList<FieldConfiguration>&)’:
../app/projectwizard.cpp:64:19: warning: ‘static QgsVectorFileWriter::WriterError QgsVectorFileWriter::writeAsVectorFormatV2(QgsVectorLayer*, const QString&, const QgsCoordinateTransformContext&, const QgsVectorFileWriter::SaveVectorOptions&, QString*, QString*, QString*)’ is deprecated [-Wdeprecated-declarations]
   64 |     &errorMessage );
      |                   ^
In file included from ../app/projectwizard.cpp:17:
../../QGIS/src/core/qgsvectorfilewriter.h:695:63: note: declared here
  695 |     Q_DECL_DEPRECATED static QgsVectorFileWriter::WriterError writeAsVectorFormatV2( QgsVectorLayer *layer,
      |                                                               ^~~~~~~~~~~~~~~~~~~~~
../app/projectwizard.cpp:64:19: warning: ‘static QgsVectorFileWriter::WriterError QgsVectorFileWriter::writeAsVectorFormatV2(QgsVectorLayer*, const QString&, const QgsCoordinateTransformContext&, const QgsVectorFileWriter::SaveVectorOptions&, QString*, QString*, QString*)’ is deprecated [-Wdeprecated-declarations]
   64 |     &errorMessage );
      |                   ^
In file included from ../app/projectwizard.cpp:17:
../../QGIS/src/core/qgsvectorfilewriter.h:695:63: note: declared here
  695 |     Q_DECL_DEPRECATED static QgsVectorFileWriter::WriterError writeAsVectorFormatV2( QgsVectorLayer *layer,
      |          
```

--- a/app/projectwizard.cpp
+++ b/app/projectwizard.cpp
@@ -58,18 +58,16 @@ QgsVectorLayer *ProjectWizard::createGpkgLayer( QString const &projectDir, QList
   options.driverName = QStringLiteral( "GPKG" );
   options.layerName = layerName;
   options.fileEncoding = QStringLiteral( "UTF-8" );
-  std::unique_ptr< QgsVectorFileWriter > writer( QgsVectorFileWriter::create( projectGpkgPath, predefinedFields, Qgis::WkbType::PointZ, layerCrs, QgsCoordinateTransformContext(), options ) );
-
 
   QString errorMessage;
-  QgsVectorFileWriter::writeAsVectorFormatV2(
+  QgsVectorFileWriter::writeAsVectorFormatV3(
     layer,
     projectGpkgPath,
     layer->transformContext(),
     options,
+    &errorMessage,
     nullptr,
-    nullptr,
-    &errorMessage );
+    nullptr );
 
   // Check and configure layer
   QgsVectorLayer *l = new QgsVectorLayer( projectGpkgPath, layerName, "ogr" );

